pandas/io/parsers.py

```diff
@@ -1065,6 +1065,38 @@ def _evaluate_usecols(usecols, names):
     return usecols
 
 
+def _validate_usecols(usecols, names):
+    """
+    Validates that all usecols are present in a given
+    list of names. If not, raise a ValueError that
+    shows what usecols are missing.
+
+    Parameters
+    ----------
+    usecols : iterable of usecols
+        The columns to validate are present in names.
+    names : iterable of names
+        The column names to check against.
+
+    Returns
+    -------
+    usecols : iterable of usecols
```

Comment:
there is a validate_usecols_arg function, is there a reason you are creating a new one?

Reply:
Sure, was going from @gfyoung's suggestion.
We could extend the existing function, at the moment it's only taking `usecols` as an arg so would be extending it's arguments as well as logic.
I've not checked if every call to `validate_usecols_arg` would have a `names` argument to pass through, so may need to default it? Although I would think we'd always have column names to check against, right?

Let me know if this is a 'must-have' for you and I'll implement.

Reply:
it certainly can be an optional argument
having 2 functions do similar things is confusing